src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java

```diff
@@ -184,6 +184,133 @@ void should_save_task_and_get_task_from_reportable_task() {
                 });
     }
 
+    @ParameterizedTest
+    @CsvSource(value = {
+        "LEGAL_OPERATIONS,Legal Operations",
+        "CTSC,CTSC",
+        "JUDICIAL,Judicial",
+        "ADMINISTRATOR,Admin",
+        "ADMIN,Admin",
+        "TEST,TEST"
+    })
+    void should_save_task_and_get_transformed_role_category_label_from_reportable_task(
+        String taskRoleCategory, String reportableTaskRoleCategoryLabel) {
+        TaskResource taskResource = buildTaskResource();
+        taskResource.setRoleCategory(taskRoleCategory);
+        TaskResource savedTaskResource = taskResourceRepository.save(taskResource);
+
+        await().ignoreException(AssertionFailedError.class)
+            .pollInterval(1, SECONDS)
+            .atMost(10, SECONDS)
+            .until(
+                () -> {
+                    List<ReportableTaskResource> reportableTaskList
+                        = miReportingServiceForTest.findByReportingTaskId(savedTaskResource.getTaskId());
+
+                    assertFalse(reportableTaskList.isEmpty());
+                    assertEquals(1, reportableTaskList.size());
+                    assertEquals(savedTaskResource.getTaskId(), reportableTaskList.get(0).getTaskId());
+                    assertEquals(reportableTaskRoleCategoryLabel, reportableTaskList.get(0).getRoleCategoryLabel());
+                    return true;
+                });
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+        "PRIVATELAW,Private Law",
+        "CIVIL,Civil",
+        "IA,Immigration and Asylum",
+        "PUBLICLAW,Public Law",
+        "TEST,TEST"
+    })
+    void should_save_task_and_get_transformed_jurisdiction_label_from_reportable_task(
+        String taskJurisdiction, String reportableTaskJurisdictionLabel) {
+        TaskResource taskResource = buildTaskResource();
+        taskResource.setJurisdiction(taskJurisdiction);
+        TaskResource savedTaskResource = taskResourceRepository.save(taskResource);
+
+        await().ignoreException(AssertionFailedError.class)
+            .pollInterval(1, SECONDS)
+            .atMost(10, SECONDS)
+            .until(
+                () -> {
+                    List<ReportableTaskResource> reportableTaskList
+                        = miReportingServiceForTest.findByReportingTaskId(savedTaskResource.getTaskId());
+
+                    assertFalse(reportableTaskList.isEmpty());
+                    assertEquals(1, reportableTaskList.size());
+                    assertEquals(savedTaskResource.getTaskId(), reportableTaskList.get(0).getTaskId());
+                    assertEquals(reportableTaskJurisdictionLabel, reportableTaskList.get(0).getJurisdictionLabel());
+                    return true;
+                });
+    }
```

Comment:
Probably could have consolidated some of the tests together since they are doing something very similar?